File: syno_backend/src/main/java/com/syno_back/backend/dto/UpdateUserCardDto.java
package com.syno_back.backend.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

import javax.validation.constraints.Size;
import java.time.LocalDateTime;
import java.util.List;

/**
 * DTO for receiving update info for <code>DbUserCard</code>
 */
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Builder
@ToString
public class UpdateUserCardDto {
    /**
     * Unique id for user card
     */
    @JsonProperty("pin")
    private String pin;

    /**
     * Actual translated word
     */
    @Size(max=100)
    @JsonProperty("translated_word")
    private String translatedWord;

    /**
     * Time when card was created
     */
    @JsonProperty("time_created")
    private LocalDateTime timeCreated;


    /**
     * Time when card was modified
     */
    @JsonProperty("time_modified")
    private LocalDateTime timeModified;

    /**
     * List of updated translation
     */
    @JsonProperty("translations")
    private List<UpdateUserTranslationDto> translations;
}

File: syno-mobile/syno-mobile/Core/Dtos/UpdateUserCardDto.swift
import Foundation

/// DTO for updating `DbUserCard` on server
class UpdateUserCardDto: Codable {
    let pin: String
    
    let translatedWord: String
    
    let timeCreated: Date?
    
    let timeModified: Date?
    
    let translations: [UpdateUserTranslationDto]
    
    /**
      Creates new `UpdateUserCardDto`
     - Parameters:
        - pin: `DbUserCard` unique id
        - translatedWord:  Card's translated word
        - timeCreated: time when card was created
        - timeModified: time when card was modified
        - translations: list of card's translations
     */
    init(pin: String, translatedWord: String, timeCreated: Date?, timeModified: Date?, translations: [UpdateUserTranslationDto]) {
        self.pin = pin
        self.translatedWord = translatedWord
        self.timeCreated = timeCreated
        self.timeModified = timeModified
        self.translations = translations
    }
}
